server/controllers/calls.js

```diff
@@ -194,17 +229,27 @@ export function recordAttempt(req, res) {
               return putCallAttempt(call_id, outcome, notes)
                 .then(() => {
                   const attempt_num = parseInt(call.attributes.attempt_num, 10);
-                  if (responses) {
-                    Promise.all(responses.map((resp) => {
-                      const { question_id, response } = resp;
-                      const responseParams = { call_id, question_id, response };
-                      return responsesService.saveNewResponse(responseParams);
-                    }))
-                    .then(() => afterPut(res, outcome, contact_id, attempt_num, campaign_id))
-                    .catch(err => console.log('error saving responses in recordAttempt function of calls controller when saving a call attempt :', err));
-                  }
-                  return afterPut(res, outcome, contact_id, attempt_num, campaign_id)
-                    .then().catch(err => console.log('error creating new call in log for required subsequent contact after recording attempt in recordAttempt functino of calls controller: ', err));
+                  return isLastAttemptedCall(campaign_id, outcome)
```

Comment:
There is a way to redo and simplify this entire logic chain by using agnostic promise chaining that lets you branch and rejoin the flow without duplicating code, but doesn't have to be a part of this PR. Just let me know if you're curious about what that might look like.

Reply:
I'm curious! Will follow-up with you on this.